### PR TITLE
Initial merge develop into master

### PR DESCRIPTION
**Pull request by Kennric**
### TODO's -
- [x] Fix `test_authentication_no_token_in_response` (only test that fails)
- [x] Add travis file
- [x] Run entire code base through rubocop
- [ ] Port `Pymesync Tests` having patch decorators

### CODE REVIEW

**Comment by iCHAIT** [comment]
### TODO's - 

- [ ] Add travis file
- [ ] Run Rubocop on entire code base

File: test_mock_rimesync.rb [hunk 1]
```diff
@@ -0,0 +1,629 @@
+require 'test/unit'
+require_relative 'rimesync'
+
+# rubocop:disable ClassLength
+class TestMockRimeSync < Test::Unit::TestCase # :nodoc:
+  def setup
+    baseurl = 'http://ts.example.com/v0'
+    @ts = TimeSync.new(baseurl, token=nil, test = true)
+    @ts.authenticate('testuser', 'testpassword', 'password')
```
Comment (subnomo) [inline test_mock_rimesync.rb]: You need to provide the names for the keyword args, like so: `@ts.authenticate(username: 'testuser', password: 'testpassword' ...`

File: test_mock_rimesync.rb [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+  end
+
+  def test_mock_authenticate
+    @ts.instance_variable_set(:@token, nil)
+    assert_equal(@ts.authenticate('example', 'ex', 'password'), Hash['token' => 'TESTTOKEN'])
```
Comment (subnomo) [inline test_mock_rimesync.rb]: Same thing here, provide names for keyword args.

**Comment by iCHAIT** [comment]
@subnomo Fixed.

File: rimesync.gemspec [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  s.required_ruby_version     = '>= 1.9.3'
+
+  s.add_dependency 'bcrypt', "= 3.1.11"
+  s.add_dependency 'rest-client', "= 1.8.0"
+  s.add_dependency 'webmock', "= 2.1.0"
+
+  s.files         = `git ls-files`.split("\n")
+  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
+  # s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
```
Comment (Kennric) [inline rimesync.gemspec]: just delete this line

File: rimesync.gemspec [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Gem::Specification.new do |s|
+  s.name        = "rimesync"
+  s.version     = '0.1.0'
+  s.licenses    = ['Apache-2']
+  s.authors     = ["OSU Open Source Lab"]
```
Comment (Kennric) [inline rimesync.gemspec]: add yourself to the list of authors

**Comment by Kennric** [comment]
+1

**Pull request merged**